src/web/index.js

```diff
@@ -278,6 +281,8 @@ if (!params.theme) {
   // The add-on may never answer, so start the loader delay.
   startLoaderDelay();
 } else {
+  // trigger a message to fetch and apply the current theme that the browser already has.
+  postMessage("fetchTheme");
```

Comment:
There is no guarantee that the extension is listening at this point, nor that the extension will reply with "fetchedTheme" before the hello/pong message. As a result, the fix here is no reliable.

In #866, I suggested to add a new flag (to the `setTheme` and possibly `addImages` call), so that the extension's background.js can choose to ignore the theme assignment if the user has already set a theme before. Have you tried that?

Reply:
Oh, I didn't quite understand that the first time. So by adding a flag you meant something like:
`postMessage("setTheme", { theme: theme, userHasTheme: true/false })` 
and then I do a condition to check that flag inside `messageHandlers` in `background.js` before applying the theme?
Is that what you meant?
Sorry and thanks!

Reply:
Yes exactly. And I would call the flag something like `defaultFromWeb: true`, and add a comment of why this was done (maybe based on the description that I wrote in the original issue).